tests/e2e/helpers/docsifyInit.js

```diff
@@ -0,0 +1,131 @@
+    contentMarkdown: '',
+    coverpageMarkdown: '',
+    debug: false,
+    docsifyURL: '/lib/docsify.js',
+    navbarMarkdown: '',
+    routes: [[]],
+    script: '',
+    scriptURLs: [],
+    sidebarMarkdown: '',
+    style: '',
+    styleURLs: ['/lib/themes/vue.css'],
+    url: `${serverURL}/docsify-init.html`,
+    waitForSelector: '#main',
+  };
+  const settings = {
+    ...defaults,
+    ...options,
+  };
+
+  // Routes
+  if (settings.routes.length > 0 && typeof settings.routes[0] === 'string') {
+    settings.routes = [settings.routes];
+  }
+  settings.routes.forEach(async ([urlGlob, response]) => {
+    if (urlGlob && response) {
+      if (typeof response === 'string') {
+        response = {
+          body: stripIndent`${response}`,
+        };
+      }
+      await page.route(urlGlob, route => route.fulfill(response));
+    }
+  });
+
+  // Docsify markdown
+  [
+    [settings.contentMarkdown, '**/README.md'],
+    [settings.coverpageMarkdown, '**/_coverpage.md', { coverpage: true }],
+    [settings.navbarMarkdown, '**/_navbar.md', { loadNavbar: true }],
+    [settings.sidebarMarkdown, '**/_sidebar.md', { loadSidebar: true }],
+  ].forEach(async ([markdown, urlGlob, config]) => {
+    if (markdown) {
+      if (config) {
+        Object.assign(settings.config, config);
+      }
+      await page.route(urlGlob, route =>
+        route.fulfill({
+          contentType: 'text/markdown',
+          body: stripIndent`${markdown}`,
+        })
+      );
+    }
+  });
+
+  // Load URL and wait for 'load' event
+  await page.goto(settings.url);
+  await page.waitForLoadState();
+
+  // Docsify configuration
+  if (Object.keys(settings.config).length > 0) {
+    await page.evaluate(config => {
+      Object.assign(window.$docsify, config);
+    }, settings.config);
+  }
+
+  // CSS (must resolve all before proceeding)
+  await Promise.all(settings.styleURLs.map(url => page.addStyleTag({ url })));
+  if (settings.style) {
+    await page.evaluate(data => {
+      const headElm = document.querySelector('head');
+      const styleElm = document.createElement('style');
+
+      styleElm.textContent = data;
+      headElm.appendChild(styleElm);
+    }, stripIndent`${settings.style}`);
+  }
+
+  // JavaScript (must load/resolve sequentially to guaraantee execution order)
+  settings.scriptURLs.push(settings.docsifyURL);
+  for (const url of settings.scriptURLs) {
+    await page.addScriptTag({ url });
+  }
+  if (settings.script) {
+    await page.evaluate(data => {
+      const headElm = document.querySelector('head');
+      const scriptElm = document.createElement('script');
+
+      scriptElm.textContent = data;
+      headElm.appendChild(scriptElm);
+    }, stripIndent`${settings.script}`);
+  }
+
+  // Docsify "Ready"
+  if (settings.waitForSelector) {
+    await page.waitForSelector(settings.waitForSelector);
+  }
+
+  // Debug
+  if (settings.debug) {
+    await jestPlaywright.debug();
+  }
+
+  return Promise.resolve();
```

Comment:
This seems to achieve mostly the same thing as just loading a `.html` file would. 

Wouldn't it be simpler to to just load an HTML file from the `live-server`, for example, then run the tests based on that? I think it would be easier to understand.

Reply:
Or to load the HTML content from a template string, for example?

Not pushing back on merging though.  The usage of `docsifyInit` on the outside is simple. Just give it some content, then test the markdown output for example. It's only on the inside where it seems complex with manually handling browser routes, etc.

Reply:
@trusktr --

An HTML file is being loaded. By default, `docsifyInit()` loads [`docsify-init.html`](https://github.com/docsifyjs/docsify/blob/jest-playwright/tests/e2e/fixtures/docsify-init.html)) and then modifies it based on the settings you provide. If you prefer to load your own HTML file, that's fine too. The goal is remove the need to create a large number of file fixtures that vary slightly from each other to test, for example, a single configuration option. If you want prefer file fixtures, you're free to use them as well. If you prefer to skip `docsifyInit()` altogether and just use Playwright comments like `await page.goTo(`{serverURL}/path/to/my-index.html`), that's fine too.